Userland/Libraries/LibVideo/DecoderError.h

```diff
@@ -21,6 +21,7 @@ using DecoderErrorOr = ErrorOr<T, DecoderError>;
 enum class DecoderErrorCategory : u32 {
     Unknown = 0,
     IO,
+    Memory,
```

Comment:
This could go in the previous commit? Or it might even be nicer to have a complete separate commit that adds the `DecoderError.h` file, and then in a subsequent commit actually use the new functionality.

Reply:
Hmm, I was thinking it made sense to add the classes of errors as they're used so that you could see their usage in the same commit.

I've added a few more "error" types (some are not exactly errors for decoders) since this PR, and I'm not sure if it may be necessary to add some more on top.

I could definitely split this up into multiple commits if necessary, though.

Reply:
Leaving this as-is if that's all right.